flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/source/IcebergTableSource.java

```diff
@@ -111,6 +118,32 @@ private DataStream<RowData> createDataStream(StreamExecutionEnvironment execEnv)
         .build();
   }
 
+  private DataStreamSource<RowData> createFLIP27Stream(StreamExecutionEnvironment env) {
+    SplitAssignerType assignerType = readableConfig.get(FlinkConfigOptions.TABLE_EXEC_SPLIT_ASSIGNER_TYPE);
+    IcebergSource<RowData> source = IcebergSource.forRowData()
+        .tableLoader(loader)
+        .assignerFactory(assignerType.factory())
+        .properties(properties)
+        .project(getProjectedSchema())
+        .limit(limit)
+        .filters(filters)
+        .flinkConfig(readableConfig)
+        .build();
+    DataStreamSource stream = env.fromSource(
+        source, WatermarkStrategy.noWatermarks(), source.name(), TypeInformation.of(RowData.class));
+
+    if (source.getBoundedness() == Boundedness.BOUNDED) {
+      int parallelism = SourceUtil.inferParallelism(readableConfig, limit, () -> source.planSplitsForBatch().size());
```

Comment:
the expensive lambda function of `planSplitsForBatch` will only be executed if config is enabled